### ISSUE TITLE
filter.load() for table column throws an internal error after table.clearFilters and filter.apply(originalCriteria)

### ISSUE THREAD

**Issue by harri-rajala-koivu-solutions** [OfficeDev/office-js#3030]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

filter.load() for table column throws an internal error after table.clearFilters and restoring original filter using filter.apply(originalCriteria).

## Your Environment
* Platform [PC desktop, Mac, iOS, Office on the web]: PC Desktop
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: Version 2210 (Build 15726.20202)
* Operating System: Windows 11
 
## Expected behavior
filter.load() function should not throw an exception.

## Current behavior
filter.load() function throws an internal error, and filter selections in the filter dialog are wrong.

### Step 1: Run onStep1 Excel JS code
This creates six tables.

### Step 2: Manual step. Select filters manually.
**!!! IMPORTANT NOTE !!!**
This must be done manually. Open the filter menu from the Filter column and select/unselect values.

* **BooleanTable**: **unselect** TRUE, **select** FALSE
* **BooleanTableWithBlank**: **unselect** TRUE, **select** FALSE and blanks
* **NumberTable**: **unselect** 999, **select** 777 and 888
* **NumberTableWithBlanks**: **unselect** 999, **select** 777, 888 and blanks
* **TextTable**: **unselect** zzz, **select** xxx and yyy
* **TextTableWithBlank**: **unselect** zzz, **select** xxx, yyy and blanks

### Step 3: Run onStep3 Excel JS code
After this step, open the filter dialog from all tables and see selected and unselected values.

**NOT OK**
* **BooleanTable**: FALSE is also unselected. 
* **BooleanTableWithBlanks**: FALSE is also unselected. 
* **NumberTable**: 777 and 888 are also unselected.

**OK**
* **NumberTableWithBlanks**
* **TextTable**
* **TextTableWithBlank**

### Step 4: Run onStep4 Excel JS code
In this step, you can see that filter.load() from the **BooleanTable** and **BooleanTableWithBlanks** throws an internal error.

```typescript
const addTable = async (context: Excel.RequestContext, info: string[][], tableIndex: number, tableName: string, filterColumnName: string, values: any[][]) => {
    const startRow = 1 + (tableIndex - 1) * 14;
    const sheet = context.workbook.worksheets.getActiveWorksheet();
    const table = sheet.tables.add("A" + (startRow + 3) + ":B" + (startRow + 3), true);
    table.name = tableName;
    table.getHeaderRowRange().values = [[tableName, filterColumnName]];
    table.rows.add(undefined, values);
    table.getRange().format.autofitColumns();
    sheet.getRange("A" + startRow + ":A" + (startRow + 2)).values = info;
    await context.sync();
};

const clearAndRestoreColumnFilter = async (context: Excel.RequestContext, tableName: string, tableIndex: number, stepIndex: number) => {
    const sheet = context.workbook.worksheets.getActiveWorksheet();
    const table = sheet.tables.getItem(tableName);
    const resultRow = tableIndex * 14 - 3 + stepIndex;
    const filterResponseRange = sheet.getRange("A" + resultRow);
    await context.sync();
    const filter = table.columns.getItem(2).filter.load();
    try {
        await context.sync();
    }
    catch (error) {
        filterResponseRange.values = [["ERROR!!! " + JSON.stringify(error)]];
        return;
    }
    // Get the original filter criteria so that we can restore that later.
    const originalCriteria = { ...filter.criteria };
    filterResponseRange.values = [[JSON.stringify(originalCriteria)]];
    // Clear all table filters.
    table.clearFilters();
    // Do something to table rows.
    console.log("Do something to table", tableName);
    // Restore the original filter criteria.
    filter.apply(originalCriteria);
    await context.sync();
};

const onStep1 = async () => {
    await Excel.run(async (context: Excel.RequestContext) => {
        const okInfo = [
            ["OK: Clear all table filters and restore the original filter works."],
            ["After Step 3, open the filter dialog and see that all filters are ok."],
            ["After Step 4, open the filter dialog and see that all filters are ok."],
        ];
        const notOkInfo1 = [
            ["NOT OK: Clear all table filters and restore the original filter do not work."],
            ["After Step 3, open the filter dialog and see that FALSE is also unselected."],
            ["After Step 4, filter.load() throws an exception."],
        ];
        const notOkInfo2 = [
            ["NOT OK: Clear all table filters and restore the original filter do not work."],
            ["After Step 3, open the filter dialog and see that FALSE is also unselected."],
            ["After Step 4, filter.load() throws an exception."],
        ];
        const notOkInfo3 = [
            ["NOT OK: Clear all table filters and restore the original filter do not work."],
            ["After Step 3, open the filter dialog and see that 777 and 888 are also unselected."],
            ["After Step 4, filter.load() works okay, but 777 and 888 are still unselected in the filter dialog."],
        ];
        await addTable(context, notOkInfo1, 1, "BooleanTable", "Boolean Filter (unselect TRUE, select FALSE)", [
            ["a1", true],
            ["a2", true],
            ["a3", true],
            ["a4", false],
            ["a5", false],
            ["a6", false],
        ]);
        await addTable(context, notOkInfo2, 2, "BooleanTableWithBlank", "Boolean Filter (unselect TRUE, select FALSE and blanks)", [
            ["a1", true],
            ["a2", true],
            ["a3", true],
            ["a4", null],
            ["a5", false],
            ["a6", false],
        ]);
        await addTable(context, notOkInfo3, 3, "NumberTable", "Number Filter (unselect 999, select 777 and 888)", [
            ["a1", 777],
            ["a2", 777],
            ["a3", 888],
            ["a4", 888],
            ["a5", 999],
            ["a6", 999],
        ]);
        await addTable(context, okInfo, 4, "NumberTableWithBlank", "Number Filter (unselect 999, select 777, 888 and blanks)", [
            ["a1", 777],
            ["a2", 777],
            ["a3", 888],
            ["a4", null],
            ["a5", 999],
            ["a6", 999],
        ]);
        await addTable(context, okInfo, 5, "TextTable", "Text Filter (unselect zzz, select xxx and yyy)", [
            ["a1", "xxx"],
            ["a2", "xxx"],
            ["a3", "yyy"],
            ["a4", "yyy"],
            ["a5", "zzz"],
            ["a6", "zzz"],
        ]);
        await addTable(context, okInfo, 6, "TextTableWithBlank", "Text Filter (unselect zzz, select xxx, yyy and blanks)", [
            ["a1", "xxx"],
            ["a2", "xxx"],
            ["a3", "yyy"],
            ["a4", null],
            ["a5", "zzz"],
            ["a6", "zzz"],
        ]);
    });
};

const onStep3 = async () => {
    await Excel.run(async (context: Excel.RequestContext) => {
        await clearAndRestoreColumnFilter(context, "BooleanTable", 1, 1);
        await clearAndRestoreColumnFilter(context, "BooleanTableWithBlank", 2, 1);
        await clearAndRestoreColumnFilter(context, "NumberTable", 3, 1);
        await clearAndRestoreColumnFilter(context, "NumberTableWithBlank", 4, 1);
        await clearAndRestoreColumnFilter(context, "TextTable", 5, 1);
        await clearAndRestoreColumnFilter(context, "TextTableWithBlank", 6, 1);
    });
};

const onStep4 = async () => {
    await Excel.run(async (context: Excel.RequestContext) => {
        await clearAndRestoreColumnFilter(context, "BooleanTable", 1, 2);
        await clearAndRestoreColumnFilter(context, "BooleanTableWithBlank", 2, 2);
        await clearAndRestoreColumnFilter(context, "NumberTable", 3, 2);
        await clearAndRestoreColumnFilter(context, "NumberTableWithBlank", 4, 2);
        await clearAndRestoreColumnFilter(context, "TextTable", 5, 2);
        await clearAndRestoreColumnFilter(context, "TextTableWithBlank", 6, 2);
    });
};
```

## Context
Our Excel Add-in modifies table data. The steps are following.
1. Use filter.load() to get the current filter criteria (originalCriteria).
2. Clear all table filters using table.clearFilters().
3. Do modifications to table data.
4. Restore the original filter criteria using filter.apply(originalCriteria)

Because of this bug, our Excel Add-in does not work.

**Comment by harri-rajala-koivu-solutions** [comment]
What is the status of this issue? Did you able to reproduce this issue?
